agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemExporter.java

```diff
@@ -126,6 +166,53 @@ CompletableResultCode internalSendByConnectionString(
     return telemetryPipeline.send(byteBuffers, connectionString, listener);
   }
 
+  private TelemetryItem createOtelResourceMetric(
+      String connectionString, Map<String, String> existingTags) {
+    MetricTelemetryBuilder builder = MetricTelemetryBuilder.create(_OTELRESOURCE_, 0);
+    builder.setConnectionString(connectionString);
+
+    Map<String, String> tags = new HashMap<>();
+    tags.put(
+        ContextTagKeys.AI_CLOUD_ROLE.toString(),
+        existingTags.get(ContextTagKeys.AI_CLOUD_ROLE.toString()));
+    tags.put(
+        ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString(),
+        existingTags.get(ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString()));
+    tags.put(
+        ContextTagKeys.AI_INTERNAL_SDK_VERSION.toString(),
+        existingTags.get(ContextTagKeys.AI_INTERNAL_SDK_VERSION.toString()));
+    tags.forEach((k, v) -> builder.addTag(k, v));
```

Comment:
probably easier to add these tags directly to the builder, instead of adding to HashMap and then to builder